RecoLocalTracker/SiPixelRecHits/interface/PixelCPEBase.h

```diff
@@ -85,10 +85,10 @@ class PixelCPEBase : public PixelClusterParameterEstimator
    
    struct ClusterParam
    {
-      ClusterParam(const SiPixelCluster & cl) : theCluster(&cl), loc_trk_pred(0.0,0.0,0.0,0.0),
-      probabilityX_(0.0), probabilityY_(0.0), probabilityQ_(0.0), qBin_(0.0),
-      isOnEdge_(false), hasBadPixels_(false), spansTwoROCs_(false), hasFilledProb_(false) {}
```

Comment:
This is intented! initialization is very (very) expensive (I can show you in the assembly the huge amount of "mov" instruction generated). The quantity not initialized are filled just after creation.  In general in CMSSW we are just shuffling bytes, not doing computation, we need to reduce that.